test/framework/k8s_utils.go

```diff
@@ -310,3 +315,71 @@ func NewClientFromServiceAccount(ctx context.Context, k8sClient kubernetes.Inter
 			}),
 	)
 }
+
+// DeployRootPod deploys a pod with root permissions for testing purposes.
+func DeployRootPod(ctx context.Context, c client.Client, namespace string, nodename *string) (*corev1.Pod, error) {
+	podPriority := int32(0)
+	allowedCharacters := "0123456789abcdefghijklmnopqrstuvwxyz"
+	id, err := gardenerutils.GenerateRandomStringFromCharset(3, allowedCharacters)
+	if err != nil {
+		return nil, err
+	}
+
+	rootPod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("rootpod-%s", id),
+			Namespace: namespace,
+			Annotations: map[string]string{
+				"kubernetes.io/psp": "gardener.privileged",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "root-container",
+					Image: "busybox",
+					Command: []string{
+						"sleep",
+						"10000000",
+					},
+					Resources:                corev1.ResourceRequirements{},
+					TerminationMessagePath:   "/dev/termination-log",
+					TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+					ImagePullPolicy:          corev1.PullAlways,
```

Comment:
Not that it'll make a big difference, but out of curiosity: is there a special need for `PullAlways`?

Reply:
No, I'll change it. Thanks